test/patterns/grep.chpl

```diff
@@ -0,0 +1,74 @@
+
+/*
+ A unix like grep function implemented in chapel. 
+*/
+
+use Path;
+use IO;
+use Regexp;
+use FileSystem;
+use SysBasic;
+use Time; 
+
+// To keep track of line numbers and speed parallel speedup
+var lineNum: int = 1;
+var clock: Timer;
+
+// Main 
+proc main(args: [] string) {
+
+  if args.size < 3 {
+    // clock.start();
+    advancedGrep(args[1]);   
+    // clock.stop();
+    // writeln(clock.elapsed());  uncomment for runtime
+  }  
+  if args.size == 3 {
+    fileGrep(args[1], args[2]);
+  }
+}
+
+
+/*
+ Simple version: needs
+   - file name
+   - string to be searched for
+*/
+proc fileGrep(toFind: string, file: string) {
+
+  var errors: syserr;
+  var toRead = open(errors, file, iomode.r);
+  var r = toRead.reader();
+
+  // Use regex library to compile into searchable regex
+  var regEx = compile(toFind);
+  var line:string;
+  var lineNum = 1;
+
+  // Search while handling errors
+  while(r.readline(line, errors)) {
```

Comment:
I don't think it's necessarily faster, but you could compute the positions in the file of the matches and then separately read the line. Probably most reasonable to leave it the way you had it for now though.